src/elements/content-uploader/progressCellRenderer.js

```diff
@@ -65,6 +71,9 @@ export default () => ({ rowData }: Props) => {
         case STATUS_STAGED:
             return <ItemProgress {...rowData} />;
         case STATUS_ERROR:
+            if (Browser.isSafari() && code === ERROR_CODE_UPLOAD_BAD_DIGEST && file.name.indexOf('.zip') !== -1) {
```

Comment:
So this is for `.zip` files and not `.pkg` files?

Reply:
There are two kinds of "package" files in macOS. The installer type, which are suffixed with `.pkg`, and a File System concept which is essentially a directory with a plist and UTI in macOS that says it should be treated as a single file. This deals with the latter.

In our case, Safari knows that Keynote files are supposed to be a single file, while other browsers do not. Safari does some magic to convert the upload into a zip file.